http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -906,33 +906,36 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       dbAppOpt <- KubernetesServiceDbQueries.getFullAppByName(msg.project, msg.appId).transaction
       dbApp <- F.fromOption(dbAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
       zone = ZoneName("us-central1-a")
-      deletePostgresDisk = deleteGalaxyPostgresDiskOnlyInGoogle(msg.project,
-                                                                zone,
-                                                                msg.appName,
-                                                                dbApp.app.appResources.namespace.name)
-        .adaptError {
-          case e =>
-            PubsubKubernetesError(
-              AppError(e.getMessage, ctx.now, ErrorAction.DeleteApp, ErrorSource.PostgresDisk, None),
-              Some(msg.appId),
-              false,
-              None,
-              None
-            )
-        }
+      deletePostgresDisk = if (dbApp.app.appType == AppType.Galaxy)
+        deleteGalaxyPostgresDiskOnlyInGoogle(msg.project, zone, msg.appName, dbApp.app.appResources.namespace.name)
+          .adaptError {
+            case e =>
+              PubsubKubernetesError(
+                AppError(e.getMessage, ctx.now, ErrorAction.DeleteApp, ErrorSource.PostgresDisk, None),
+                Some(msg.appId),
+                false,
+                None,
+                None
+              )
+          }
+      else F.unit
 
       deleteDisksInParallel = List(deleteDisk, deletePostgresDisk).parSequence_
 
-      // The app must be deleted before the nodepool and disk, to future proof against the app potentially flushing the postgres db somewhere
       task = for {
-        //we record the last disk detach timestamp here, before it is removed from galaxy
-        originalDiskOpt <- googleDiskService.getDisk(
-          msg.project,
-          zone,
-          gkeInterp.getGalaxyPostgresDiskName(dbApp.app.appResources.namespace.name)
-        )
-        originalDetachTimestampOpt = originalDiskOpt.map(_.getLastDetachTimestamp)
+        // For Galaxy apps, wait for the postgres disk to detach before deleting the disks
+        // TODO should we do this for the other disk too?
```

Comment:
I'd think we'd want to do this for the other disk too. Am I missing anything?

Reply:
The old logic only checked the postgres disk, so I kept that consistent. But yes it feels like we'd need to wait for all disks to detach.

@Qi77Qi's PR has more changes to disk logic - maybe that could be done there?

Reply:
yea..less change in this PR will be desirable :) it's going to cause merge conflicts as is and these can be tricky to resolve

For the original question, if we've seen instability around this, I'm all for adding the detaching check; otherwise, I'm thinking it might be ok to leave as is 🤷‍♀️

Reply:
just fyi, I did see one occurrence of disk deletion failure in my local test, so I'm going to add the wait for data disk as well